snapcraft/parts/setup_assets.py

```diff
@@ -33,23 +33,35 @@
 
 
 def setup_assets(
-    project: models.Project, *, assets_dir: Path, project_dir: Path, prime_dir: Path
+    project: models.Project,
+    *,
+    assets_dir: Path,
+    project_dir: Path,
+    prime_dir: Path,
+    meta_directory_handler: Optional[Callable[[Path, Path], None]] = None,
```

Comment:
I appreciate the solid typing here and the docstring explaining how the handler is called.